kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+)
+
+func (d *deployer) Up() error {
+	if err := d.verifyFlags(); err != nil {
+		return fmt.Errorf("up could not verify flags: %s", err)
+	}
+
+	if err := d.enableComputeAPI(); err != nil {
+		return fmt.Errorf("up couldn't enable compute API: %s", err)
+	}
+
+	env := d.buildEnv()
+	script := filepath.Join(d.RepoRoot, "cluster", "kube-up.sh")
+
+	cmd := exec.Command(script)
+	cmd.SetEnv(env...)
+	exec.InheritOutput(cmd)
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("error encountered during %s: %s", script, err)
+	}
+
+	return nil
+}
+
+func (d *deployer) buildEnv() []string {
+	env := []string{}
+
+	// path is necessary for scripts to find gsutil, gcloud, etc
+	env = append(env, fmt.Sprintf("PATH=%s", os.Getenv("PATH")))
+	env = append(env, fmt.Sprintf("PROJECT=%s", d.GCPProject))
+	env = append(env, fmt.Sprintf("KUBECONFIG=%s", d.kubeconfigPath))
+
+	return env
+}
+
+func (d *deployer) verifyFlags() error {
+	if err := d.setRepoPathIfNotSet(); err != nil {
+		return err
+	}
+
+	if d.GCPProject == "" {
+		return fmt.Errorf("GCP Project must be set.")
+	}
+
+	return nil
+}
+
+func (d *deployer) enableComputeAPI() error {
+	// In freshly created GCP projects, the compute API is
+	// not enabled. We need it. Enabling it after it has
+	// already been enabled is a relatively fast no-op,
+	// so this can be called without consequence.
+
+	env := os.Environ()
+	env = append(env, fmt.Sprintf("CLOUDSDK_CORE_PROJECT=%s", d.GCPProject))
+
+	cmd := exec.Command(
```

Comment:
optonal nit: using the golang API would be awesome here, but i know we rely on gcloud to be existing for kube-up anyway.

Reply:
Agreed. However, I couldn't find services as a supported component of the golang API when I looked.

Reply:
It seems I was looking in the wrong place, or at least in the "recommended new library" that doesn't have this. Tried using it but it looks like I need to resolve project name to project number first (https://cloud.google.com/service-usage/docs/enable-disable#curl, https://cloud.google.com/resource-manager/docs/creating-managing-projects#api_1). Given the necessary existence of `gcloud` and with this being the only place we interact with the API (for now), I think its fine as is. When the deployer has to start creating firewall rules, etc. It may be worth revisiting.